source/MissionAction.h

```diff
@@ -81,8 +82,11 @@ class MissionAction {
 	std::string system;
 	LocationFilter systemFilter;
 
+	// Condition under which the dialog Decline button is visible
+	std::shared_ptr<ConditionSet> toDecline;
```

Comment:
If a person wants conditions to influence how a mission is offered to the player, they should use a conversation.

Reply:
> If a person wants conditions to influence how a mission is offered to the player, they should use a conversation.

Removing the "decline" button is for situations where the player should *not* have the ability to influence the mission. Also, it is not limited to "on offer" dialogs.

We need this for three reasons:

1. User interface consistency. The stopover and blocked dialogs are not conversations. Other informational messages should be as well.
2. A conversation with six words and a "done" looks sloppy. A dialog with an "ok" button does not. This is why some missions already use dialogs.
3. We can do this now, but the player can decline the mission in a situation where it makes no sense to do so.

Hence, the need for a "to decline" condition. I could use a simple "no decline" flag, but a condition makes it more powerful, and is close to the same amount of code.